clang/include/clang/CIR/Dialect/IR/CIRTypes.h

```diff
@@ -19,6 +19,43 @@
 
 #include "clang/CIR/Interfaces/ASTAttrInterfaces.h"
 
+//===----------------------------------------------------------------------===//
+// CIR FloatType
+//
+// The base type for all floating-point types.
+//===----------------------------------------------------------------------===//
+
+namespace mlir {
+namespace cir {
+
+class SingleType;
+class DoubleType;
+
+class FloatType : public Type {
```

Comment:
Any reason this definition is in a header file? Seems like the relevant methods can be shared with an interface.

Reply:
OK I'll move this to an interface.

Reply:
Updated.